libbeat/common/version_test.go

```diff
@@ -45,6 +45,61 @@ func TestVersion(t *testing.T) {
 	}
 }
 
+func TestEqual(t *testing.T) {
+	tests := []struct {
+		name     string
+		version  string
+		version1 string
+		result   bool
+	}{
+		{
+			name:     "5.4.3 < 6.4.3",
+			version:  "5.4.3",
+			version1: "6.4.3",
+			result:   false,
+		},
+		{
+			name:     "5.7.3 < 5.8.3",
+			version:  "5.7.3",
+			version1: "5.8.2",
+			result:   false,
+		},
+		{
+			name:     "5.4.3 > 5.4.2",
+			version:  "5.4.3",
+			version1: "5.4.2",
+			result:   false,
+		},
+		{
+			name:     "5.4.3 = 5.4.3",
+			version:  "5.4.3",
+			version1: "5.4.3",
+			result:   true,
+		},
+		{
+			name:     "1.2.3 = 1.2.3-alpha1",
```

Comment:
Ok, I see why comment above would not work. I would not treat these as equal (at least in our stack as there are breaking changes between each alpha release.